Tasks/AzureRmWebAppDeployment/task.json

```diff
@@ -15,28 +15,39 @@
     "author": "Microsoft Corporation",
     "version": {
         "Major": 3,
-        "Minor": 2,
-        "Patch": 4
+        "Minor": 3,
+        "Patch": 0
     },
     "releaseNotes": "What's new in Version 3.0: <br/>&nbsp;&nbsp;Supports File Transformations (XDT) <br/>&nbsp;&nbsp;Supports Variable Substitutions(XML, JSON) <br/>Click [here](https://aka.ms/azurermwebdeployreadme) for more Information.",
     "minimumAgentVersion": "2.104.1",
-    "groups": [{
+    "groups": [
+        {
             "name": "FileTransformsAndVariableSubstitution",
             "displayName": "File Transforms & Variable Substitution Options",
-            "isExpanded": false
+            "isExpanded": false,
+            "visibleRule": "WebAppKind = app || WebAppKind = functionapp"
         },
         {
             "name": "AdditionalDeploymentOptions",
             "displayName": "Additional Deployment Options",
-            "isExpanded": false
+            "isExpanded": false,
+            "visibleRule": "WebAppKind = app || WebAppKind = functionapp"
         },
         {
             "name": "output",
             "displayName": "Output",
-            "isExpanded": true
+            "isExpanded": true,
+            "visibleRule": "WebAppKind = app || WebAppKind = functionapp"
+        },
+        {
+            "name": "ApplicationSettings",
+            "displayName": "Application Settings",
+            "isExpanded": true,
+            "visibleRule": "WebAppKind = linux"
```

Comment:
This change is fine for now. Later when we support built-in linux deployment, this option's visibility should be controlled.